modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/ProfileUpgradeBannerViewModel.kt

```diff
@@ -105,4 +133,50 @@ class ProfileUpgradeBannerViewModel @Inject constructor(
             settings.setLastSelectedSubscriptionTier(upgradeFeatureCard.subscriptionTier)
         }
     }
+
+    private fun getDefaultSubscription(
+        filteredSubscriptions: List<Subscription>,
+        cachedTier: SubscriptionTier?,
+        cachedSubscriptionStatus: SubscriptionStatus?,
+    ) = subscriptionManager.getDefaultSubscription(
+        subscriptions = filteredSubscriptions,
+        tier = if (cachedTier == SubscriptionTier.PATRON) {
+            Subscription.SubscriptionTier.PATRON
+        } else {
+            Subscription.SubscriptionTier.PLUS
+        },
+        frequency = getSubscriptionFrequency(cachedSubscriptionStatus)
+    )
+
+    private fun getSubscriptionFrequency(cachedSubscriptionStatus: SubscriptionStatus?) =
+        if (cachedSubscriptionStatus is SubscriptionStatus.Paid) {
+            when (cachedSubscriptionStatus.frequency) {
+                SubscriptionFrequency.NONE -> SubscriptionFrequency.YEARLY
+                SubscriptionFrequency.MONTHLY -> SubscriptionFrequency.MONTHLY
+                SubscriptionFrequency.YEARLY -> SubscriptionFrequency.YEARLY
+            }
+        } else {
+            SubscriptionFrequency.YEARLY
+        }
+
+    private fun getPlanType(
+        productTier: Subscription.SubscriptionTier,
+        cachedTier: SubscriptionTier?,
+        cachedSubscriptionStatus: SubscriptionStatus?,
+    ): PlanType {
+        val productTierMatchesUserSubscriptionTier =
+            productTier.name.lowercase() == cachedTier?.label?.lowercase()
+        val isExpiring = if (productTierMatchesUserSubscriptionTier) {
+            (cachedSubscriptionStatus as? SubscriptionStatus.Paid)?.isExpiring ?: false
+        } else {
+            false
+        }
```

Comment:
Instead of `else { false }` we could convert this into an `&&` with something like:

``` kotlin
val isExpiring = productTierMatchesUserSubscriptionTier &&
    (cachedSubscriptionStatus as? SubscriptionStatus.Paid)?.isExpiring == true
```

I find this a bit easier to follow the logic, but what you have is fine as well—it's just another option. Feel free to leave as-is

Reply:
Updated in https://github.com/Automattic/pocket-casts-android/commit/a6138221c0f11b64e2b4b2825ac032ffa4cd0b2f for more clarity